code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -288,3 +317,266 @@
+
+	var/datum/action/xeno_action/stealth/S = locate() in X.xeno_abilities //Gotta reference the datum for whether we're stealthed/able to sneak attack
+
+	if(!S) //Sanity; should not be possible.
+		to_chat(X, "<span class='xenowarning'>We somehow don't have the ability to stealth!</span>")
+		return FALSE
+
+	if(!S.stealth || !S.can_sneak_attack)
+		to_chat(X, "<span class='xenowarning'>We must be able to sneak attack the target to properly sting it!</span>")
+		return FALSE
+
+	return TRUE
+
+
+/datum/action/xeno_action/activable/sneak_stinger/on_cooldown_finish()
+	var/mob/living/carbon/xenomorph/X = owner
+	to_chat(X, "<span class='xenowarning'><b>We can now sting our victims again as our toxin glands refill.</b></span>")
+	X.playsound_local(X.loc, 'sound/voice/alien_drool1.ogg', 25, 0, 1)
+	return ..()
+
+
+/datum/action/xeno_action/activable/sneak_stinger/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/carbon/C = A
+
+	X.face_atom(C)
+
+	to_chat(X, "<span class='xenodanger'>We prepare to sting our quarry...</span>")
+
+	if(!do_after(X, HUNTER_STINGER_WINDUP, TRUE, target, BUSY_ICON_HOSTILE)) //Slight wind up
+		return fail_activate()
+
+	C.attack_alien(X) //We auto-sneak attack as part of this ability.
+	var/datum/reagent/toxin = /datum/reagent/toxin/acid/xeno_acid
+	var/transfer_amount = HUNTER_SNEAK_ATTACK_INJECT_AMOUNT
+	if(X.a_intent != INTENT_HARM) //Inject neurotoxin instead of acid while on disarm intent
+		toxin = /datum/reagent/toxin/xeno_neurotoxin
+		transfer_amount *= 0.5 //We inject only half the regular amount for neurotoxin
+
+	C.reagents.add_reagent(toxin, transfer_amount)
+	to_chat(C, "<span class='danger'>You feel a tiny prick.</span>") //Fluff
+	to_chat(X, "<span class='xenowarning'>Our stinger silently injects our victim!</span>")
+	X.playsound_local(C, 'sound/effects/spray3.ogg', 5, 0, 1)
+	C.playsound_local(C, 'sound/effects/spray3.ogg', 5, 0, 1)
+
+	succeed_activate()
+
+	GLOB.round_statistics.hunter_stings++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "hunter_stings") //Statistics
+	add_cooldown()
+
+
+// ***************************************
+// *********** Hunter's Mark
+// ***************************************
+/datum/action/xeno_action/activable/hunter_mark
+	name = "Hunter's Mark"
+	action_icon_state = "hunter_mark"
+	mechanics_text = "Psychically mark a creature you have line of sight to, allowing you to sense its direction, distance and location."
+	plasma_cost = HUNTER_STINGER_PLASMA_COST
+	keybind_signal = COMSIG_XENOABILITY_HUNTER_MARK
+	cooldown_timer = HUNTER_MARK_COOLDOWN
+	var/mob/hunter_mark_target
+
+/datum/action/xeno_action/activable/hunter_mark/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	if(QDELETED(A))
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/hunter/X = owner
+	var/mob/C = A
+
+	if(!istype(C))
+		to_chat(X, "<span class='xenowarning'>We cannot psychically mark this target!</span>")
```

Comment:
A) use isliving
B) typecast it straightaway in the proc and just check it later so we dont have to reassign it

Reply:
I want to make Hunter's Mark and Psychic Trace as general as possible

Reply:
Anyways this is done.